pallets/accounts/src/lib.rs

```diff
@@ -476,89 +523,102 @@ pub mod pallet {
 			let pub_key = &pub_key[12..];
 			ensure!(pub_key == sender.0, Error::<T>::VerificationFailed);
 
-			let amount: u128 = data.amount.try_into().map_err(|_| Error::<T>::SystematicFailure)?;
+			let data_amount: u128 = data.amount.try_into().map_err(|_| Error::<T>::SystematicFailure)?;
 
 			if !data.lock {
-				ensure!(amount == 0, Error::<T>::SystematicFailure);
+				ensure!(data_amount == 0, Error::<T>::SystematicFailure);
 			} else {
-				ensure!(amount > 0, Error::<T>::SystematicFailure);
+				ensure!(data_amount > 0, Error::<T>::SystematicFailure);
 			}
 
 			// verifications ended, let's proceed with voting count and writing
 
 			let threshold = T::Threshold::get();
-			if threshold > 1 {
-				// 问Gio
-				let current_votes = <EVMLinkVoting<T>>::get(&data_hash);
-				if let Some(current_votes) = current_votes {
-					// Number of votes for the key `data_hash` in EVMLinkVoting
-					if current_votes + 1u64 < threshold {
-						// Current Votes has not passed the threshold
-						<EVMLinkVoting<T>>::insert(&data_hash, current_votes + 1);
-						return Ok(());
-					} else {
-						// Current votes passes the threshold, let's remove EVMLinkVoting perque perque non! (问Gio)
-						// we are good to go, but let's remove the record
-						<EVMLinkVoting<T>>::remove(&data_hash);
-					}
-				} else {
-					// If key `data_hash` doesn't exist in EVMLinkVoting
-					<EVMLinkVoting<T>>::insert(&data_hash, 1);
-					return Ok(());
-				}
-			}
+			let _ = Self::vote_count_and_write(&data_hash, threshold);
```

Comment:
This is a red flag, why result is ignored?

Reply:
Oh, right. I will change it to `Self::vote_count_and_write(&data_hash, threshold)?;` .